admin/main.js

```diff
@@ -1,85 +1,76 @@
-      fetch(url, {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/json'
-          },
-          body: JSON.stringify(data)
-      }).then(response => {
-        response.json()
-        if (response.ok) {
-          location.reload(true);
-        };
-        console.log(response)
-        return response
-      } )     //  return response.json();
+    return fetch(url, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(data)
+    });
   }
-  fetch('/admin/rpc/list-users').then((response) => {
-    response.json().then(resp => {
-      for (const user of resp) {
-        const li = document.createElement('li');
-        const button = document.createElement('button');
-        const spanName = document.createElement('span');
-        const spanButton = document.createElement('span');
-        const strong = document.createElement('strong');
-        button.innerText ='x';
-        button.className = 'list'
-        spanButton.className = 'button'
-        spanName.innerText = user.name + ' ' + user.lastName;
-        li.appendChild(spanName);
-        if(username !== user.username) {
-          spanButton.appendChild(button);
-          li.appendChild(spanButton);
-          list.appendChild(li);
-          
-          button.addEventListener("click",  () => {
-            postData('/admin/rpc/delete-users', {
-                userID: user.userID
-            });
-          });
-        } else {
-          strong.innerText ='Admin';
-          spanButton.appendChild(strong);
-          li.appendChild(spanButton);
-          list.appendChild(li);
-          
-          button.addEventListener("click",  () => {
-            postData('/admin/rpc/delete-users', {
-                userID: user.userID
-            });
-          });
-        }
+
+  function refresh() {
+    list.innerHTML = '';
+    for (const user of users) {
+      const li = document.createElement('li');
+      const spanName = document.createElement('span');
+      const spanButton = document.createElement('span');
+      spanButton.className = 'button';
+      spanName.innerText = user.name + ' ' + user.lastName;
+      li.appendChild(spanName);
+      if (username !== user.username) {
+        spanButton.innerHTML = `<button class="list" data-id="${user.userID}">x</button>`;
+      } else {
+        spanButton.innerText = 'Admin';
       }
-    }, error => {
-      console.log(error);
-    });
+      li.appendChild(spanButton);
+      list.appendChild(li);
+    }
+  }
+
+  list.addEventListener('click',  (event) => {
+    const target = event.target;
+    if (target.classList.contains('list')) {
+      const userID = parseFloat(target.getAttribute('data-id'));
+      postData('/admin/rpc/delete-users', {
+        userID: userID
+      }).then(() => {
+        users = users.filter(user => userID !== user.userID);
```

Comment:
See how in this occasion I remove the user from the list. So I change my copy of the array of users.
When we add a user instead of adding it to the list I reload the list from the server. both options are fine with up and downside. The upside of doing it in place as we are doing it here is that we do not hit the server and the database, the change is immediate. the downside is that if we make an error the list of users in the server and the list of users in the webpage may diverge.
Consider what would happen, for instance if some other admin is also adding a user at the same time.

Reply:
I prefer to hit the server and database in order to prevent a discrepancy, or at least I would like to compare the new local list with the database updated to confirm that all is ok